rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -27,6 +27,33 @@ namespace rosbag2_cpp
 {
 namespace readers
 {
+namespace details
+{
+std::vector<std::string> resolve_relative_paths(
+  const std::string & base_folder, std::vector<std::string> relative_files, size_t version = 4)
+{
+  auto base_path = rcpputils::fs::path(base_folder);
+  if (version < 4) {
+    // In older rosbags (version <=3) relative files are prefixed with the rosbag folder name
+    base_path = rcpputils::fs::path(base_folder).parent_path();
+  }
+
+  rcpputils::require_true(base_path.exists(), "base folder does not exist: " + base_folder);
+  if (!base_path.is_directory()) {
+    throw std::invalid_argument("base folder has to be a directory: " + base_folder);
+  }
```

Comment:
```suggestion
  rcpputils::require_true(base_path.is_directory(), "base folder has to be a directory: " + base_folder);
```
I accidentally left this suggestion out of the previous review.

Reply:
I've addressed this locally as well ;-)